TTYG-178 Improve documentation (#51)
Ontotext-AD/graphrag-eval: docs/steps-score.md

# Steps score

The steps score is an overall metric of the correctness of the steps the chat
bot or agent executed in response to the user's query. It can be used to find

> the understand how to improve the agent.

Reviewer: `the understand` is not grammatically correct. My suggestion is to phrase this as `It can be used to understand how to improve the agent.`

Author: Thanks! Fixed.

The steps score is a real number in the interval [0, 1], 

> which indicates how closely the actual steps match to the reference steps.

Reviewer: ```suggestion
which indicates how closely the actual steps match the reference steps.
```

Author: Fixed

in order: actual steps that match earlier groups must be executed before all 
steps matching later groups.
- If some steps in the current reference group are not matched, then the 
matching stops, and the score is computed from the matches found so far.

An actual step can match at most one reference step, and only if the actual 
step is successful (i.e., it didn’t result in an error). There are a few ways 
a reference step can match an actual one. In all cases except for 
the "retrieval" steps, the matching score is either 0 or 1. A score above 0 
indicates a match.

- if both are named "sparql_query" and the "output_media_type" of the 
reference step is "application/sparql-results+json", then we try to match them 
using the [SPARQL queries comparison algorithm](#sparql-queries-comparison).
- if both are named "retrieval" and the reference step has "output", then we 
compute [recall@k](retrieval-evaluation-using-chunk-ids.md#recall-k).
- if both are named "retrieve_time_series", then we check if the arguments of 
the steps are matching.
- if both are named "retrieve_data_points", then we check if the arguments of 

> the steps are matching.

Reviewer: ```suggestion
- if both are named "retrieve_time_series", then we check if the arguments of 
the steps match.
- if both are named "retrieve_data_points", then we check if the arguments of 
the steps match.
```

Author: Fixed

- if the reference step is named "iri_discovery" and the actual step name is 

> "autocomplete_search", тhen check if the IRI specified as "output" of the 

Reviewer: ```suggestion
"autocomplete_search", then we check if the IRI specified as "output" of the 
```

Author: How about if we remove the "we" from all the bullet points, e.g.:

```
- if both are named "sparql_query" and the "output_media_type" of the 
reference step is "application/sparql-results+json", then try to match them 
using the [SPARQL queries comparison algorithm](#sparql-queries-comparison).
...
```

Reviewer: I'm fine with this, the focus of the suggestion was to fix the garbled world `тhen`

Author: Fixed and revamped

where scores range between 0 and 1. A score greater than 0 indicates at least
a partial match.

Matching of a reference step to an actual step works as follows:

- if both steps are named "sparql_query" and the "output_media_type" of the 
reference step is "application/sparql-results+json", then compare the steps
using [SPARQL queries comparison](#sparql-queries-comparison).
- if both steps are named "retrieval" and the reference step has "output", then
compute [recall@k](retrieval-evaluation-using-chunk-ids.md#recall-k).
- if both are named "retrieve_time_series", then check if the arguments of 
the steps match.
- if both are named "retrieve_data_points", then check if the arguments of 
the steps match.
- if the reference step is named "iri_discovery" and the actual step name is 
"autocomplete_search", then check if the IRI specified as "output" of the 
"iri_discovery" step is present in the "output" of the "autocomplete_search".
- if the reference and actual step names are the same and the 
"output_media_type" of the reference step is "application/json", and the json
outputs are the same then the steps match.
- match the outputs of the two steps.

## SPARQL queries comparison

> To compare a an actual SPARQL query to a reference SPARQL query, we compare

Reviewer: ```suggestion
To compare an actual SPARQL query to a reference SPARQL query, we compare
```

Author: Fixed

their results. We do this as follows:

- If the reference result has $n$ columns, consider all subsets of
$n$ columns in actual result

> - For each subset, compare each column to each reference colum (skipping

Reviewer: ```suggestion
- For each subset, compare each column to each reference column (skipping
```

Author: Fixed
